Source/ResourceLibUnitTests/AcceleratorResourceTests.cs

```diff
@@ -32,5 +32,65 @@ public void TestLoadAcceleratorResources()
                 Assert.AreEqual(110, ri[Kernel32.ResourceTypes.RT_ACCELERATOR][1].Name.Id.ToInt64());
             }
         }
+
+        [Test]
+        public void TestAccelerator()
+        {
+            string filename = Path.Combine(Environment.GetEnvironmentVariable("WINDIR"), "explorer.exe");
+
+            Predicate<Accelerator> pReload = (Accelerator a) => { return a.Key == "VK_F5" && a.Command == 41061; };
+            using (ResourceInfo ri = new ResourceInfo())
+            {
+                ri.Load(filename);
+                foreach (AcceleratorResource rc in ri[Kernel32.ResourceTypes.RT_ACCELERATOR])
+                {
+                    Console.WriteLine("Current AcceleratorResource: {0}, {1}", rc.Name, rc.TypeName);
+                    Console.WriteLine(rc + "\n------------------------------------\n");
+
+                    Accelerator newAC = new Accelerator();
+                    newAC.Key = "J";
+                    newAC.Command = 41008;
+                    newAC.Flags = User32.AcceleratorVirtualKey.VIRTKEY | User32.AcceleratorVirtualKey.NOINVERT | User32.AcceleratorVirtualKey.CONTROL;
+
+                    try
+                    {
+                        Accelerator tst = rc.Accelerators.Find(pReload);
+                        int loc = rc.Accelerators.FindIndex(pReload);
+                        rc.Accelerators.Remove(tst);
+
+                        tst.Key = "VK_NUMPAD9";
+                        tst.addFlag(User32.AcceleratorVirtualKey.CONTROL);
+                        tst.removeFlag(User32.AcceleratorVirtualKey.ALT | User32.AcceleratorVirtualKey.NOINVERT);
+                        rc.Accelerators.Insert(loc, tst);
+                        rc.Accelerators.Add(newAC);
+
+                        List<Accelerator> acList = new List<Accelerator>();
+
+                        Accelerator acA = new Accelerator();
+                        acA.Command = 413;
+                        acList.Add(acA);
+                        acList[0].addFlag(User32.AcceleratorVirtualKey.SHIFT);
+                        acList[0].Key = "VK_F1";
+
+                        Accelerator acB = new Accelerator();
+                        acB.Flags = User32.AcceleratorVirtualKey.VIRTKEY | User32.AcceleratorVirtualKey.NOINVERT;
+                        acB.Key = "Z";
+                        acList.Add(acB);
+
+                        rc.Accelerators.AddRange(acList);
+                        Console.WriteLine("Modified AcceleratorResource: " + rc.Name);
+                        Console.WriteLine(rc);
+                        Console.ReadLine();
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine("Error: " + e.ToString());

```

Comment:
For a unit test, you want to "assert" things, not have a pass-through fail like this. Look at the `TestLoadAcceleratorResources` above for an example. Maybe read up a bit on NUnit.

I think you want to break this up in multiple tests.

* Adding one accelerator.
* Modifying one accelerator.
* Removing one accelerator.
* Doing something on multiple accelerators.

Think of `TestLoadAcceleratorResources` as a start. Now that there's a lot more functionality in accelerators supported write the smallest tests possible for each of those functions. Together they cover the entire functionality of the code that you're adding, across multiple tests.